### ISSUE TITLE
UCSC MySQL mirror selection

### ISSUE THREAD

**Issue by gokceneraslan** [vanheeringen-lab/genomepy#239]
Hi,

I am having issues with running `genomepy install hg38` via the API within a Gitlab CI/CD environment on our internal gitlab instance although it works fine locally:

![image](https://github.com/vanheeringen-lab/genomepy/assets/1140359/da006206-faed-4492-9536-eccccd2240f5)

I am wondering if there is a way to specify a MySQL mirror for USCS genome downloads such as adding an option to use `genome-euro-mysql.soe.ucsc.edu` here:

https://github.com/vanheeringen-lab/genomepy/blob/d641c7adf9c9d7d04175773ead393a8008d6d630/genomepy/providers/ucsc.py#L523

since it's the alternative mysql server:

https://genome.ucsc.edu/goldenPath/help/mysql.html

Or maybe the code can try the Europe server if the US server fails. Let me know what you think.

**Comment by siebrenf** [comment]
Hey @gokceneraslan,

this could be a nice feature, but rather situational. The difference between your local machine and the CI server is most likely the firewall. Changing the mirror probably won't help in that case :(

It could still be nice to switch to the EU mirror in case the US mirror is down. This could be done fairly easily for installing data, but for downloading the UCSC assembly metadata it would be less easy. I'll think a bit about this...

**Comment by gokceneraslan** [comment]
Thank you!

**Comment by gokceneraslan** [comment]
Another related question: is it normal to see the same error even when we call `genomepy.install_genome(annotation=False)`? Because what we want is just the genome sequence, not annotations.

**Comment by siebrenf** [comment]
I've made it into a config option in #242. Not super elegant, but it works :) 

try it with
```
pip install git+https://github.com/vanheeringen-lab/genomepy.git@ucsc
genomepy config generate
```
and edit the `ucsc_mirror` in the config file.

And to your other question: the MySQL server is always queried to populate the UCSC metadata. So if you still get the error with this PR, it's very likely a firewall issue.

**Comment by siebrenf** [comment]
genomepy uses MySQL for UCSC starting from version 10, so you could also try an older version.

**Comment by siebrenf** [comment]
This feature is now available in genomepy 0.16!
